sgv2-service-common/src/main/java/io/stargate/sgv2/common/grpc/DefaultStargateBridgeClientFactory.java

```diff
@@ -15,34 +15,30 @@
  */
 package io.stargate.sgv2.common.grpc;
 
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
 import io.grpc.Channel;
+import io.stargate.proto.Schema.CqlKeyspaceDescribe;
 import io.stargate.proto.Schema.SchemaRead;
 import java.util.Optional;
-import java.util.concurrent.ScheduledExecutorService;
 
 class DefaultStargateBridgeClientFactory implements StargateBridgeClientFactory {
 
   private final Channel channel;
   private final SchemaRead.SourceApi sourceApi;
-  private final DefaultStargateBridgeSchema schema;
+  private final Cache<String, CqlKeyspaceDescribe> keyspaceCache =
+      Caffeine.newBuilder()
+          // TODO tune max size, TTL...
+          .maximumSize(10000)
```

Comment:
So this is global cache, shared between all instances? 10K max keyspaces.. Yea we should tune this, this can be a lot of memory imo -> 1GB if keyspace in average takes 100KB.. Not sure what the average is..

Reply:
If and when caffeine caches allow specifying weight of items, could use some simple heuristics for calculating weight for cached entries.

Could also consider ttl (in hours probably) to evict least-recently used entries for cases where cache is not full by entry count.

Reply:
I did a few tests regarding `CqlKeyspaceDescribe` size:
- 20 tables with 20 columns/table retains 63KB.
- 200 tables with 75 columns/table (the Astra limits) retains 2.2MB.

200 tables is a lot, and even then not all of them are going to have that many columns. I think an order of magnitude of a few hundred KBs is realistic. So a cache size of 1000 sounds appropriate.

Reply:
Yup, data never lies.. Maybe we can add some extra protection with soft values/keys.. Just to avoid really hard fail here with OOM..